R/dbi-driver.R

```diff
@@ -48,6 +48,10 @@ setMethod("show", "OdbcDriver",
 #'   not using UTF-8 you will need to set the encoding to get accurate
 #'   re-encoding. See [iconvlist()] for a complete list of available encodings
 #'   on your system. Note strings are always returned `UTF-8` encoded.
+#' @param columnNameEncoding The text encoding for column names used on the
+#'   Database.  May be different than the `encoding` argument.  Defaults to
+#'   empty string which is equivalent to returning the column names without
```

Comment:
I'm slightly surprised that this doesn't default to `encoding`. Are the column names and values encoded differently that often?

Reply:
Hi Hadley:

The issue that precipitated this PR is an instance where the content is encoded `latin1` but the names are `UTF-8`.  Prior to 1.5.0 the names were never de/coded regardless how how the `encoding` argument was specified, so in a sense we are just reverting to prior behavior, while retaining the flexibility that @shrektan added in 1.5.0.